### PR TITLE
Create some unit tests for ready.go

### PR DESCRIPTION
**Pull request by asalkeld**

### CODE REVIEW

**Comment by asalkeld** [comment]
This is failing:

Openshift on Azure end user e2e tests [EndUser]
/go/src/github.com/openshift/openshift-azure/test/e2e/end_user_test.go:16
  should disallow PDB mutations [It]
  /go/src/github.com/openshift/openshift-azure/test/e2e/end_user_test.go:33

  Expected
      <bool>: false
  to equal
      <bool>: true

  /go/src/github.com/openshift/openshift-azure/test/e2e/end_user_test.go:49
------------------------------
SS

Summarizing 1 Failure:

[Fail] Openshift on Azure end user e2e tests [EndUser] [It] should disallow PDB mutations 
/go/src/github.com/openshift/openshift-azure/test/e2e/end_user_test.go:49

File: pkg/upgrade/ready_test.go [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+		},
+	}
+	for _, tt := range tests {
+		if got := isPodReady(tt.pod); got != tt.want {
+			t.Errorf("isPodReady(%s) = %v, want %v", tt.name, got, tt.want)
+		}
+	}
+}
+
+func TestNodeReady(t *testing.T) {
```
Comment (jim-minter) [inline pkg/upgrade/ready_test.go]: yikes, we have a nodeIsReady function and a isNodeReady function :(
can you refactor isNodeReady into nodeIsReady and condense the testing?

File: pkg/upgrade/ready_test.go [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+	}
+}
+
+func TestNodeIsReady(t *testing.T) {
+	tests := []struct {
+		name     string
+		kc       kubernetes.Interface
+		nodeName string
+		want     bool
+		wantErr  bool
```
Comment (jim-minter) [inline pkg/upgrade/ready_test.go]: can you easily test that if the clientset returns some other error, it propagates?

File: pkg/upgrade/ready_test.go [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+	}{
+		{
+			name:     "not found",
+			nodeName: "toad",
```
Comment (jim-minter) [inline pkg/upgrade/ready_test.go]: s/toad/master-000000/ might be kinder

File: pkg/upgrade/ready_test.go (hunk 2 above)
Comment (asalkeld) [reply pkg/upgrade/ready_test.go]: I have tried using AddReactor, but it doesn't seem to be hitting.

**Pull request merged**